internal/orchestrator/updates.go

```diff
@@ -388,28 +434,34 @@ func (o *Orchestrator) eventNeedsReconciliation(evt *v1types.ConditionUpdateEven
 	})
 
 	lastTime, err := registry.LastContact(evt.ControllerID)
-	if errors.Is(err, nats.ErrKeyNotFound) {
-		return true
-	}
+	if err != nil {
+		if errors.Is(err, nats.ErrKeyNotFound) {
+			return true
+		}
 
-	if err == nil {
-		since := time.Since(lastTime)
-		le.WithFields(logrus.Fields{
-			"last.update":    evt.UpdatedAt.String(),
-			"worker.checkin": since.String(),
-		}).Debug("long running event caught by reconciler")
-	} else {
 		le.WithError(err).Warn("error checking registry")
+		return false
 	}
 
+	// we don't attempt to clean up, but collect metrics on it
+	since := time.Since(lastTime)
+	le.WithFields(logrus.Fields{
+		"last.update":    evt.UpdatedAt.String(),
+		"worker.checkin": since.String(),
+	}).Debug("long running event caught by reconciler")
+
+	metrics.ConditionReconcileStale.With(
+		prometheus.Labels{"conditionKind": string(evt.Kind)},
+	).Inc()
+
 	return false
 }
 
-func (o *Orchestrator) startReconciler(ctx context.Context, wg *sync.WaitGroup) {
-	wg.Add(1)
+func (o *Orchestrator) startReconciler(ctx context.Context) {
```

Comment:
Don't do this. I explicitly use a different `WaitGroup` on process start to handle this. Ganging it all into the member `WaitGroup` isn't needed.

Reply:
why does it have to be separate waitgroups instead of the shared, is it better practice?

Reply:
The member WaitGroup could be used by different callers, for different purposes, but they share a `Wait()` because they are the same object.

Reply:
Yep thats a fair point. 
Although in this case the reconciler routine is being spawned by the Orchestrator [directly](https://github.com/metal-toolbox/conditionorc/blob/f64bd86ecd3f4a312f1a320e399d92822e1c20b3/internal/orchestrator/orchestrator.go#L137) so it would be ideal to have it use the parent waitgroup.. or how do you propose it be done in this case?

Reply:
That imposes a burden on a maintainer to be sure that the member WaitGroup isn't actually being used for something else during system start, and it's vulnerable to someone making a change that compiles but leads to odd and inconsistent behavior at runtime that will be difficult to debug.